src/modules/payment/forms/payment-form.tsx

```diff
@@ -0,0 +1,170 @@
+import type { SubmitErrorHandler } from 'react-hook-form';
+import { FormProvider } from 'react-hook-form';
+import { z } from 'zod';
+
+import { withFormHOC } from '../../../core/hoc/with-form-hoc';
+import { common } from '../../../translations/en.json';
+import { ControlledInput } from '../../../ui/core/input';
+import { MonthYearPicker, schema as MonthYearSchema } from '../../../ui/core/month-year-picker';
+import { Text } from '../../../ui/core/text';
+import { View } from '../../../ui/core/view';
+
+export const _adressSchema = z.object({
+  city: z.string({
+    required_error: 'City is required',
+  }),
+  country: z.string({
+    required_error: 'Country is required.',
+  }),
+  adress1: z.string({
+    required_error: 'First adress is required',
+  }),
+  adress2: z
+    .string({
+      required_error: 'Second adress is required',
+    })
+    .optional(),
+  postalCode: z.string({
+    required_error: 'Postal code is required',
+  }),
+});
+
+const _cardSchema = z.object({
+  cardNumber: z.string({
+    required_error: 'Card number is required',
+  }),
+  cvcCode: z.string({
+    required_error: 'CVC is required',
+  }),
+});
+
+const schema = _adressSchema.and(MonthYearSchema).and(_cardSchema);
+
+export type TBuyFormFields = z.infer<typeof schema>;
+
+export type BuyFormSubmitHandler = (fields: TBuyFormFields) => void;
+export type BuyFormSubmitErrorHandler = SubmitErrorHandler<TBuyFormFields>;
+
+const BuyForm = withFormHOC(
+  {
+    schema,
+    config: {
+      shouldFocusError: false,
+      mode: 'onSubmit',
+    },
+  },
+  ({ formMethods, disabled }) => {
+    const { control } = formMethods;
+
+    return (
+      <>
+        <View className="flex-1">
+          <Text variant="h6-bold" className="my-4">
+            {common.labels.shippingAdress}
+          </Text>
+          <Text variant="h6" className="mb-1.5">
+            {common.labels.city}
+            {common.labels.starSign}
+          </Text>
+          <ControlledInput
+            control={control}
+            name="city"
+            placeholder={common.labels.city}
+            editable={!disabled}
+          />
+        </View>
+        <View className="flex-1">
+          <Text variant="h6" className="mb-1.5 mt-3">
+            {common.labels.country}
+            {common.labels.starSign}
+          </Text>
+          <ControlledInput
+            control={control}
+            name="country"
+            placeholder={common.labels.country}
+            editable={!disabled}
+          />
+        </View>
```

Comment:
Same comment from other form file applies, a formfield component should be definitely created. Seems like the formfields could also be rendered on your FormHOC based on a list of fields you can pass through properties, and the title could also be rendered there.

Reply:
Done